src/hotspot/share/opto/superword.cpp

```diff
@@ -749,86 +742,6 @@ int SuperWord::get_iv_adjustment(MemNode* mem_ref) {
   return iv_adjustment;
 }
 
-//---------------------------dependence_graph---------------------------
-// Construct dependency graph.
-// Add dependence edges to load/store nodes for memory dependence
-//    A.out()->DependNode.in(1) and DependNode.out()->B.prec(x)
-void SuperWord::dependence_graph() {
-  CountedLoopNode *cl = lpt()->_head->as_CountedLoop();
-  assert(cl->is_main_loop(), "SLP should only work on main loops");
-
-  // First, assign a dependence node to each memory node
-  for (int i = 0; i < body().length(); i++ ) {
-    Node* n = body().at(i);
-    if (n->is_Mem() || n->is_memory_phi()) {
-      _dg.make_node(n);
-    }
-  }
-
-  const GrowableArray<PhiNode*>& mem_slice_head = _vloop_analyzer.memory_slices().heads();
-  const GrowableArray<MemNode*>& mem_slice_tail = _vloop_analyzer.memory_slices().tails();
-
-  ResourceMark rm;
-  GrowableArray<Node*> slice_nodes;
-
-  // For each memory slice, create the dependences
-  for (int i = 0; i < mem_slice_head.length(); i++) {
-    PhiNode* head = mem_slice_head.at(i);
-    MemNode* tail = mem_slice_tail.at(i);
-
-    // Get slice in predecessor order (last is first)
-    _vloop_analyzer.memory_slices().get_slice_in_reverse_order(head, tail, slice_nodes);
-
-    // Make the slice dependent on the root
-    DepMem* slice = _dg.dep(head);
-    _dg.make_edge(_dg.root(), slice);
-
-    // Create a sink for the slice
-    DepMem* slice_sink = _dg.make_node(nullptr);
-    _dg.make_edge(slice_sink, _dg.tail());
```

Comment:
Note: as far as I can tell, there is no reason to create these `sink` nodes. They are later never referenced again.